temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/RootNamespaceAutoConfiguration.java

```diff
@@ -64,11 +70,31 @@ public RootNamespaceAutoConfiguration(ConfigurableListableBeanFactory beanFactor
   public NamespaceTemplate rootNamespaceTemplate(
       TemporalProperties properties,
       WorkflowServiceStubs workflowServiceStubs,
+      @Autowired List<DataConverter> dataConverters,
```

Comment:
Why does this have to be a collection?

Reply:
It doesn't. We allow users to have several converters. If there is only one - we just use it. If there are several - we make the user specify the "one" by giving it a specific name.

Reply:
I see, it's so we can use the first one in the list if there's only 1 but fallback to main converter if there is ambiguity.


I wonder then if, instead of warn and fallback to default, we should error if there is ambiguity in custom data converters. The reason is, it may be unclear that just by adding a second data converter you are now disabling the first essentially. Someone may add a data converter in some other part of the spring app not knowing they are causing this.

Basically, we should probably require explicit data converter config/choice in the face of multiple instead of ignoring all.

Reply:
Yeah, I was debating about the same with myself and I don't have a strong preference. If you are in favor of erroring out in this situation - I will change to this approach.